Visi.Repository/VisiQueryFactory.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System;
+using Vonk.Core.Common;
+using Vonk.Facade.Relational;
+
+namespace Visi.Repository
+{
+    public abstract class VisiQueryFactory<E, Q> : RelationalQueryFactory<E, Q> where E : class where Q : RelationalQuery<E>, new()
```

Comment:
I have to admit this is black magic

Reply:
@vadi2 : Generics are from the dark side. Powerful but dangerous :-)